src-tauri/src/model/karma.rs

```diff
@@ -0,0 +1,70 @@
+use sqlx::Encode;
+use thiserror::Error;
+
+//todo: add name in karma model
+
+#[derive(Debug, Error)]
+pub enum KarmaError {
+    #[error("Could not convert {0} into a Karma Type")]
+    InvalidNumericKarmaType(i32),
+}
+
+#[derive(Debug, Clone, Encode)]
+pub struct KarmaPoint {
+    purpose: KarmaType,
+    close_type: KarmaType,
+    closed: bool,
+}
+
+impl KarmaPoint {
+    pub fn new(purpose: KarmaType) -> KarmaPoint {
+        // Initially we set the closing type to the purpose
+        // assuming it will be closed correctly, and we change that at
+        // closing time
```

Comment:
Why would we want close_type to have a value other than the purpose at the closing time?

Reply:
That is the idea of the application, to track that you are spending energy on what you are trying to achieve. E.g. when you create a karma and give it purpose=sport, at the end of the day if you did not do any sport you could for example close it with close_type=Procrastination and that will be used to compute your karma score

Reply:
That would make close_type=Procrastination a very important type. Adding it as an enum element will highlight the intention mentioned above.